ContactList/ContactList/TableViewController.swift

```diff
@@ -0,0 +1,98 @@
+//
+//  TableViewController.swift
+//  ContactList
+//
+//  Created by Maksym Humeniuk on 5/6/19.
+//  Copyright © 2019 Maksym Humeniuk. All rights reserved.
+//
+
+import UIKit
+
+class TableViewController: UITableViewController {
+    
+    var personsArray = [Person]()
+    
+    @IBOutlet var emptyListView: UIView!
+    
+    @IBOutlet weak var addNewContactButton: UIBarButtonItem!
+    var buttonCopy: UIBarButtonItem?
+    
+    @IBAction func addNewContact(_ sender: Any?) {
+        
+        let controller = self.storyboard!.instantiateViewController(withIdentifier: "ViewControllerForUpdate") as! ViewControllerForUpdate
+        controller.handler = self
+        controller.createNewPerson()
+        let navController = UINavigationController(rootViewController: controller)
+        self.present(navController, animated:true, completion: nil)
+    }
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        tableView.backgroundView = emptyListView
+        buttonCopy = addNewContactButton
+    }
+    
+    func addContactButtonSetVisibility() {
+        navigationItem.setRightBarButton(!personsArray.isEmpty ? buttonCopy : nil, animated: true)
+    }
+
+    // MARK: - Table view data source
+
+    override func numberOfSections(in tableView: UITableView) -> Int {
+        return 1
+    }
+
+    override func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        
+        if personsArray.isEmpty {
```

Comment:
`numberOfSections(in tableView: UITableView) -> Int` is more suitable place to call this